packages/main/src/plugin/task-manager.ts

```diff
@@ -28,7 +30,30 @@ export class TaskManager {
 
   private tasks = new Map<string, Task>();
 
-  constructor(private apiSender: ApiSenderType) {}
+  constructor(
+    private apiSender: ApiSenderType,
+    private statusBarRegistry: StatusBarRegistry,
+    private commandRegistry: CommandRegistry,
+  ) {}
+
+  public init() {
+    // The TaskManager is responsible for creating the entry he will be using
+    this.statusBarRegistry.setEntry(
```

Comment:
not for this PR but probably would be a nice enhancement on statusBarRegistry to return the Entry when calling setEntry method